addons/point_of_sale/models/pos_session.py

```diff
@@ -802,8 +802,9 @@ def _accumulate_amounts(self, data):
                         .filtered(lambda m: not bool(m.origin_returned_move_id and sum(m.stock_valuation_layer_ids.mapped('quantity')) >= 0))\
                         .mapped('stock_valuation_layer_ids')
                     for move in stock_moves_batch.with_context(candidates_prefetch_ids=candidates._prefetch_ids):
-                        exp_key = move.product_id._get_product_accounts()['expense']
-                        out_key = move.product_id.categ_id.property_stock_account_output_categ_id
+                        fpos = order_line.order_id.fiscal_position_id
```

Comment:
@robinengels I noticed that `order_line` is used here, but I couldn't find where it was defined earlier in the code. This could potentially lead to a scenario where `order_line` unintentionally retains the last value from a previous iteration over `orderlines`. Given that different orders may have varying fiscal positions, it's crucial to ensure that the correct fiscal position is used for each `order_line`. There's a risk that `order_line` might be undefined in certain cases, leading to errors. Could you please verify its definition and ensure it's correctly handled to avoid such issues?

Reply:
@pedrambiria Indeed you'r right. I'll look for another way to get the fiscal position

Reply:
@robinengels Please consider reverting the changes. There's an issue with a database session that cannot be closed. If an alternative solution cannot be identified before the weekly update, reverting may be necessary to avoid disruptions.

Reply:
The changes is being reverted now by this PR https://github.com/odoo/odoo/pull/179084